clang/lib/AST/Expr.cpp

```diff
@@ -263,6 +263,14 @@ namespace {
   }
 }
 
+QualType Expr::getEnumCoercedType(const ASTContext &Ctx) const {
+  bool NotEnumType = dyn_cast<EnumType>(this->getType()) == nullptr;
+  if (NotEnumType)
```

Comment:
This look redundant since `getEnumConstantDecl()` will just return `nullptr`

Reply:
In C mode enumeration constants have integral type, so `getEnumConstantDecl()` might not return `nullptr` even if the expression doesnt have enum type.